data/state/accountsDB.go

```diff
@@ -87,6 +87,34 @@ func getNumCheckpoints(trie data.Trie) uint32 {
 	return binary.BigEndian.Uint32(val)
 }
 
+func (adb *AccountsDB) GetCode(account AccountHandler) []byte {
+	baseAcc, ok := account.(baseAccountHandler)
+	if !ok {
+		return nil
+	}
+
+	if len(baseAcc.GetCode()) != 0 {
+		return baseAcc.GetCode()
```

Comment:
assuming the base account has a code inside and returning it in that case might lead to this scenario:
I have an account, I've put by mistake something in its code field and then I call AccountsDB.GetCode assuming I will get the "real" contract code. Instead I get the existing byte slice from the user account.

Reply:
Should I retrieve the code from the trie each time?

Reply:
I think it is better without this sort of cache (we can do it in other parts of the code). In this way this function express the intent in a more straight forward way.

Reply:
you can get it - as we have extra gas cost for it now.